### ISSUE TITLE
Document/provide a way of turning off metric averaging

### ISSUE THREAD

**Issue by mrahtz** [keras-team/keras#17690]
**Describe the feature and the current behavior/state**.

As I understand it, the metrics that Keras reports through eg stdout/callbacks are averaged over time steps. For example, in any given epoch, the loss reported at step 10 is not the instantaneous loss from the 10th batch - it's the average of the losses of the first 10 steps.

One significant effect of this is that the loss can jump significantly between the end of one epoch and the start of the next, when metrics appear to be reset.

I can't find any mention of this in the documentation or other GitHub issues, and there doesn't seem to be a way to disable this behaviour. It would be great to both a) make it clearer to users this is what's happening (I had trouble finding _any_ information about this online, other than the odd unsolved Stack Overflow thread like https://stackoverflow.com/q/58381152 - and only finally figured it out after about 45 minutes of digging in Keras guts), and b) give the user a way to control this behaviour, so that they can choose whether to do any loss smoothing themselves through eg TensorBoard.

**Will this change the current api? How?**

If it were made configurable, I guess it would involve an API change to something like `model.compile()`.

**Who will benefit from this feature?**

Folks who plot loss metrics through something like TensorBoard that does its own configurable smoothing.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): yes, I could give this a try, if you _are_ happy to make this configurable

**Comment by SuryanarayanaY** [comment]
Hi @mrahtz ,

You may not straight away able do this using `model.complie()` or `model.fit()`. But there is option for users to customize the training loop,loss/metrics calculation and logging using custom callbacks. Please refer to the sources attached here,[link1](https://www.tensorflow.org/api_docs/python/tf/keras/Model#compute_loss) and [link2](https://www.tensorflow.org/guide/keras/custom_callback) and [link3](https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit?hl=en).
